discourse-doctor

```diff
@@ -0,0 +1,403 @@
+  echo -e "HTTP/1.1 200 OK\n\n $VERIFY" | nc -w 4 -l -p $PORT >/dev/null 2>&1 &
+  if curl --proto =http -s $HOST:$PORT --connect-timeout 3 | grep $VERIFY >/dev/null 2>&1
+  then
+      return 0
+  else
+    curl --proto =http -s localhost:$PORT >/dev/null 2>&1
+    return 1
+  fi
+}
+
+check_IP_match () {
+  HOST="$1"
+  echo
+  echo Checking your domain name . . .
+  if connect_to_port $HOST 443
+  then
+      echo
+      echo "Connection to $HOST succeeded."
+  else
+    echo WARNING:: This server does not appear to be accessible at $HOST:443.
+    echo
+    if connect_to_port $HOST 80
+    then
+	echo A connection to port 80 succeeds, however.
+	echo This suggests that your DNS settings are correct,
+	echo but something is keeping traffic to port 443 from getting to your server.
+	echo Check your networking configuration to see that connections to port 443 are allowed.
+    else
+      echo "A connection to http://$HOST (port 80) also fails."
+      echo
+      echo This suggests that $HOST resolves to the wrong IP address
+      echo or that traffic is not being routed to your server.
+    fi
+    echo
+    echo Google: \"open ports YOUR CLOUD SERVICE\" for information for resolving this problem.
+    echo
+    echo You should probably answer \"n\" at the next prompt and disable Let\'s Encrypt.
+    echo
+    echo This test might not work for all situations,
+    echo so if you can access Discourse at http://$HOST, you might try anyway.
+    sleep 3
+  fi
+}
+
+##
+## Do we have docker?
+##
+check_docker_version () {
+  echo  -e "\n\n==================== DOCKER INFO ====================" | tee -a $LOG_FILE
+  docker_path=`which docker.io || which docker`
+  if [ -z $docker_path ]; then
+    echo "Docker is not installed. Have you installed Discourse at all?"
+    echo "Perhaps you're looking for ./discourse-setup ."
+    echo "There is no point in continuing."
+    exit
+  else
+    echo -n "DOCKER VERSION: " | tee -a $LOG_FILE
+    docker --version | tee -a  $LOG_FILE
+    echo -e "\nDOCKER PROCESSES" | tee -a $LOG_FILE
+    sudo docker ps -a | tee -a $LOG_FILE
+  fi
+}
+
+##
+## What are we running on
+##
+check_OS() {
+  echo "----------------------------------------"
+  echo -n "OS:"  |tee -a $LOG_FILE
+  echo `uname -s` |tee -a  $LOG_FILE
+}
+
+##
+## OS X available memory
+##
+check_osx_memory() {
+  echo "----------------------------------------"
+  echo -n "OSX RAM:" | tee -a $LOG_FILE
+  free -m | awk '/Mem:/ {print $2}' | tee -a  $LOG_FILE
+  free -m | tee -a $LOG_FILE
+}
+
+##
+## Linux available memory
+##
+check_linux_memory() {
+  echo "----------------------------------------"
+  echo -n "RAM (MB):" |tee -a $LOG_FILE
+  free -m --si | awk ' /Mem:/  {print $2} ' | tee -a $LOG_FILE
+  free -m | tee -a $LOG_FILE
+}
+
+##
+## Do we have enough memory and disk space for Discourse?
+##
+check_disk_and_memory() {
+
+  os_type=$(check_OS)
+  avail_mem=0
+  if [ "$os_type" == "Darwin" ]; then
+    avail_mem=$(check_osx_memory)
```

Comment:
where is avail_mem used ?

Reply:
In `discourse-setup`. I pulled out the variable.